### PR TITLE
removed unnecessary helper functions from io.ascii test file

### LINKED ISSUE #15782: Remove needless functions from an `io.ascii` test helper file
### What is the problem this feature will solve?

A helper file in `io.ascii` tests contains three functions that serve little purpose: https://github.com/astropy/astropy/blob/fefe2261fcf09751d052d80bd8143e72f18f6aea/astropy/io/ascii/tests/common.py#L27-L37

### Describe the desired outcome

The aforementioned functions should be removed and any tests still using them should be updated accordingly. Pull requests targeting one or more functions are welcome.

Potential contributors can have a look at #16644 that removed the helper functions from one file and use that as an example.

### Additional context

`assert_equal()` and `assert_true()` should be inlined, but `assert_almost_equal(a, b)` should be replaced with [`np.testing.assert_allclose(a, b)`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose) instead of `assert np.allclose(a, b)` because the dedicated testing function produces a better error message if the check fails.

### Performed removals

- [ ] `assert_equal()`
- [ ] `assert_almost_equal()`
- [ ] `assert_true()`

### PR DESCRIPTION
**Pull request by uorae**
### Description

This pull request is to address the needless functions in a helper file for io.ascii.

Fixes #15782

- assert_equal, assert_true and assert_almost_equal are removed from common.py
- all instances in test files are changed

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/changes/io.ascii/16908.bugfix.rst
Comment (pllim) [inline docs/changes/io.ascii/16908.bugfix.rst:1]: Since the changes are not user-facing (only touching test code), we do not need a change log. Please remove this file.

Comment (uorae) [reply docs/changes/io.ascii/16908.bugfix.rst:1]: I think I ticked the no squash and merge request on accident sorry, the option is fine! Just removed the change log.

**Comment by pllim** [comment]
pre-commit.ci autofix

**Review by eerovaher — APPROVED** [review]
This reduces the amount of code used for running `io.ascii` tests without making the tests any less informative. It should be good to merge if the commits are squashed, but the sub-package maintainers should do the final review.

**Review by mhvk — APPROVED** [review]
Nice! Agreed that the commits should be squashed (or a rebase done making a small set of logical commits).

**Comment by mhvk** [comment]
I see that the don't squash box is now unticked, so I'll go ahead and merge - I think this is an obvious improvement where I doubt anyone will mind not having to do further review. Thanks, @uorae!

**Comment by bsipocz** [comment]
A deprecation as a heads-up would have been nice here as this function was used downstream (and astroquery might not be the only one doing so).

**Comment by bsipocz** [comment]
I run into this again as now @pllim was complaining about that astroquery needs to have a release, practically asap. 

These explicitly part of the public API functions just simply should not have been removed without a deprecation. The deprecation pattern and policy in place for a reason, and it should have come up during the review process. It's totally not nice to forcefully make downstream to release for reasons such as this removal.

**Comment by pllim** [comment]
Well, if it only affects testing, maybe not as urgent, unless you have users who also run similar testing regularly. It was just a FYI issue. I reused the same text for other packages. Sorry for the confusion!

**Comment by bsipocz** [comment]
The "needs a new release" is not a FYI phrasing, especially if it is due to a mistake upstream has made. We do dev testing, and do RC testing and smoke out all the issues in the dev cycle, so the flipside of that would be for upstream to actually follow their own deprecation policies.

**Pull request merged**
